Source/SG/SG.cs

```diff
@@ -104,18 +123,26 @@ public static void ConfigureInstrument(NIRfsg rfsgHandle, InstrumentConfiguratio
 
             rfsgHandle.FrequencyReference.Source = RfsgFrequencyReferenceSource.FromString(instrConfig.ReferenceClockSource);
 
-            if (instrConfig.ShareLOSGToSA)
+            // Only configure LO settings on supported VSTs
+            if (Regex.IsMatch(rfsgHandle.Identity.InstrumentModel, "NI PXIe-58[34].")) // Matches 583x and 584x VST families
             {
-                rfsgHandle.RF.LocalOscillator.LOOutEnabled = true;
-                rfsgHandle.RF.LocalOscillator.Source = RfsgLocalOscillatorSource.Onboard;
-            }
-            else
-            {
-                rfsgHandle.RF.LocalOscillator.LOOutEnabled = false;
-                // Return to the default value, in case in future modifications the above case changes
-                // this to something other than the default
-                rfsgHandle.RF.LocalOscillator.Source = RfsgLocalOscillatorSource.Onboard;
+                IntPtr rfsgPtr = rfsgHandle.GetInstrumentHandle().DangerousGetHandle();
+                NIRfsgPlayback.RetrieveAutomaticSGSASharedLO(rfsgPtr, "", out RfsgPlaybackAutomaticSGSASharedLO currentMode);
+                if (instrConfig.LOSharingMode == LocalOscillatorSharingMode.None && currentMode != RfsgPlaybackAutomaticSGSASharedLO.Disabled)
+                {
+                    //Setting this property resets other settings, which can create issues. Hence, it is only set if the value
+                    //is different than the current mode.
+                    NIRfsgPlayback.StoreAutomaticSGSASharedLO(rfsgPtr, "", RfsgPlaybackAutomaticSGSASharedLO.Disabled);
+                }
+                else if (currentMode != RfsgPlaybackAutomaticSGSASharedLO.Enabled)

```

Comment:
```suggestion
                else if (instrConfig.LOSharingMode == LocalOscillatorSharingMode.Automatic && currentMode != RfsgPlaybackAutomaticSGSASharedLO.Enabled)
```

Found during unit testing. If the LOSharingMode is set to None and currentMode is Disabled, current code will enable AutoSGSAShared. This suggestion should provide a stronger check for toggling the value.